snowpack/src/build/build-import-proxy.ts

```diff
@@ -66,6 +67,20 @@ export function wrapHtmlResponse({
   // replace %PUBLIC_URL% in HTML files (along with surrounding slashes, if any)
   code = code.replace(/\/?%PUBLIC_URL%\/?/g, config.buildOptions.baseUrl);
 
+  const snowpackPublicEnv = getSnowpackPublicEnvVariables();
+
+  code = code.replace(/%SNOWPACK_PUBLIC_[a-z0-9_]+%/gi, (match: string) => {
```

Comment:
nit: I'd prefer to be less strict here and use something like `/%SNOWPACK_PUBLIC_.+?%/` since we are already performing filtering via `PUBLIC_ENV_REGEX` below. And, if it doesn't match we throw a warning already, so we should be safe from that perspective as well.

Reply:
Sounds good

```suggestion
  code = code.replace(/%SNOWPACK_PUBLIC_.+?%/gi, (match: string) => {
```